test/BF16/mlp-all.mlir

```diff
@@ -1,17 +1,19 @@
-// RUN: tpp-opt %s --pack-vnni="block-factors=2"  --one-shot-bufferize="bufferize-function-boundaries allow-return-allocs function-boundary-type-conversion=identity-layout-map" -canonicalize -drop-equivalent-buffer-results -finalizing-bufferize  --linalg-ext-to-loops --convert-linalg-to-tpp --convert-check-to-loops --convert-vnni-to-tpp --convert-tpp-to-xsmm -convert-xsmm-to-func -convert-vector-to-scf -convert-scf-to-cf  --convert-math-to-llvm --lower-affine --reconcile-unrealized-casts |\
-// RUN: tpp-run \
-// RUN:  -e entry -entry-point-result=void
+// RUN: tpp-opt %s -pack-vnni="block-factors=2"  -generalize-tensor-pack-unpack -empty-tensor-to-alloc-tensor -one-shot-bufferize="bufferize-function-boundaries allow-return-allocs function-boundary-type-conversion=identity-layout-map" -drop-equivalent-buffer-results -finalizing-bufferize -convert-vnni-to-tpp -canonicalize -convert-tpp-to-xsmm -convert-xsmm-to-func -canonicalize -convert-check-to-loops -convert-linalg-to-loops -convert-vector-to-scf -convert-scf-to-cf -expand-strided-metadata -lower-affine -convert-arith-to-llvm -convert-vector-to-llvm -convert-memref-to-llvm -arith-expand -convert-math-to-llvm -convert-func-to-llvm -reconcile-unrealized-casts |\
+// RUN: mlir-cpu-runner \
+// RUN:  -e entry -entry-point-result=void \
+// RUN: -shared-libs=%llvmlibdir/libmlir_c_runner_utils%shlibext,%tpplibdir/libtpp_c_runner_utils%shlibext
 //
-// RUN: tpp-opt %s --pack-vnni="block-factors=2"  --one-shot-bufferize="bufferize-function-boundaries allow-return-allocs function-boundary-type-conversion=identity-layout-map" -canonicalize -drop-equivalent-buffer-results -finalizing-bufferize  --linalg-ext-to-loops --convert-linalg-to-tpp --convert-vnni-to-tpp | FileCheck %s -check-prefix=TPP
+// RUN: tpp-opt %s -pack-vnni="block-factors=2" -generalize-tensor-pack-unpack -empty-tensor-to-alloc-tensor -one-shot-bufferize="bufferize-function-boundaries allow-return-allocs function-boundary-type-conversion=identity-layout-map" -canonicalize -drop-equivalent-buffer-results -finalizing-bufferize -convert-linalg-to-tpp -convert-vnni-to-tpp | FileCheck %s -check-prefix=TPP
```

Comment:
If the new generalize pass needs an alloc tensor pass right afterwards, than we may be doing something wrong. We should not use those passes unless there's no other way.

If this is just about the one `empty` tensor in main, than it could very well be a `fill`.

Reply:
generalize-tensor-pack-unpack does not emit any `tensor.empty`. The op is there because we build pack/unpack with a tensor.empty as destination. We can always allocate for tensor.pack or tensor.unpack but we need more thinking especially if you start fusion or vectorizing this ops.